website/content/docs/api-gateway/upgrade-specific-versions.mdx

```diff
@@ -0,0 +1,240 @@
+        - backendRefs:
+          - group: ""
+            kind: Service
+            name: web-backend
+            namespace: gateway-namespace
+            ...
+      ...
+  ```
+1. Inspect the `backendRefs` entries for each of the routes. 
+  
+  If a `namespace` field is not defined in the `backendRef` or if the namespace matches the namespace of the route, then no additional action is required for the `backendRef`. Otherwise, note the `group`, `kind`, `name`, and `namespace` field values for `backendRef` configurations that have a `namespace` defined that do not match the namespace of the parent route. You must also note the `kind` and `namespace` of the parent route. You will need these to create a `ReferencePolicy` that explicitly allows each cross-namespace route-to-service pair to prevent the route from breaking (see [step 5](#create-reference-policy)). 
+  
+  After completing this step, you will have a list of all routes similar to the following:
+
+  ```yaml
+    example-http-route:
+      - namespace: example-namespace
+        kind: HTTPRoute
+        backendReferences:
+          - group : ""
+            kind: Service
+            name: web-backend
+            namespace: gateway-namespace
+
+    example-tcp-route:
+      - namespace: a-different-namespace
+        kind: HTTPRoute
+        backendReferences:
+          - group : ""
+            kind: Service
+            name: web-backend
+            namespace: gateway-namespace
+  ```
+
+  Skip to [step 8](#step-8) if your list is empty. 
+<a name="jq-command"/>
+
+1.  If you have installed [`jq`](https://stedolan.github.io/jq/), issue the following command to get all `HTTPRoutes` and `TCPRoutes` and filter for routes that require a `ReferencePolicy`.
+
+  ```shell-session
+  $ kubectl get HTTPRoute,TCPRoute -o json -A | jq -r '.items[] | {name: .metadata.name, namespace: .metadata.namespace, kind: .kind, crossNamespaceBackendReferences: ( .metadata.namespace as $parentnamespace | .spec.rules[] .backendRefs[] | select(.namespace != null and .namespace != $parentnamespace )  )} '
+  ```
+  
+  Note that the command retrieves all `HTTPRoutes` and `TCPRoutes`. `TLSRoutes` and `UDPRoutes` are not supported in v0.1.0.
+
+  The output will resemble the following response if routes that require a new `ReferencePolicy` are returned:
+
+  ```log
+  {
+    "name": "example-http-route",
+    "namespace": "example-namespace",
+    "kind": "HTTPRoute",
+    "crossNamespaceBackendReferences": {
+      "group": "",
+      "kind": "Service",
+      "name": "web-backend",
+      "namespace": "gateway-namespace",
+      "port": 8080,
+      "weight": 1
+    }
+  }
+  {
+    "name": "example-tcp-route",
+    "namespace": "a-different-namespace",
+    "kind": "TCPRoute",
+    "crossNamespaceBackendReferences": {
+      "group": "",
+      "kind": "Service",
+      "name": "web-backend",
+      "namespace": "gateway-namespace",
+      "port": 8080,
+      "weight": 1
+    }
+  }
+  ```
```

Comment:
We should put this in a CodeBlock, that way we can use `hideClipboard` attribute and prevent people from thinking this a template. Otherwise, they will use the copy button and get errors.

Reply:
Looked at some other code in this repo and confirmed that you can actually hide the clipboard directly after the language specification 🥳 

```
```yaml hideClipboard 
```

Reply:
Even better ❤️